File: sample-videochat-webrtc/src/main/java/com/quickblox/sample/groupchatwebrtc/util/NetworkConnectionChecker.java
package com.quickblox.sample.groupchatwebrtc.util;

import android.app.Application;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import java.util.Set;
import java.util.concurrent.CopyOnWriteArraySet;

public class NetworkConnectionChecker {

    private final ConnectivityManager connectivityManager;

    private Set<OnConnectivityChangedListener> listeners = new CopyOnWriteArraySet<>();

    public NetworkConnectionChecker(Application context) {
        this.connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        IntentFilter intentFilter = new IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION);
        context.registerReceiver(new NetworkStateReceiver(), intentFilter);
    }

    public void registerListener(OnConnectivityChangedListener listener) {
        listeners.add(listener);
        listener.connectivityChanged(isConnectedNow());
    }

    public void unregisterListener(OnConnectivityChangedListener listener) {
        listeners.remove(listener);
    }

    public boolean isConnectedNow() {
        NetworkInfo activeNetworkInfo = connectivityManager.getActiveNetworkInfo();
        return activeNetworkInfo != null && activeNetworkInfo.isConnected();
    }

    public interface OnConnectivityChangedListener {

        void connectivityChanged(boolean availableNow);

    }

    private class NetworkStateReceiver extends BroadcastReceiver {

        @Override
        public void onReceive(Context context, Intent intent) {
            boolean isConnectedNow = isConnectedNow();

            for (OnConnectivityChangedListener listener : listeners) {
                listener.connectivityChanged(isConnectedNow);
            }
        }
    }
}

File: sample-videochat-kotlin/app/src/main/java/com/quickblox/sample/videochat/kotlin/util/NetworkConnectionChecker.kt
package com.quickblox.sample.videochat.kotlin.util

import android.app.Application
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.net.ConnectivityManager
import java.util.concurrent.CopyOnWriteArraySet


class NetworkConnectionChecker(context: Application) {

    private val connectivityManager: ConnectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
    private val listeners = CopyOnWriteArraySet<OnConnectivityChangedListener>()

    init {
        val intentFilter = IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION)
        context.registerReceiver(NetworkStateReceiver(), intentFilter)
    }

    fun registerListener(listener: OnConnectivityChangedListener) {
        listeners.add(listener)
        listener.connectivityChanged(isConnectedNow())
    }

    fun unregisterListener(listener: OnConnectivityChangedListener) {
        listeners.remove(listener)
    }

    private fun isConnectedNow(): Boolean {
        val activeNetworkInfo = connectivityManager.activeNetworkInfo
        return activeNetworkInfo != null && activeNetworkInfo.isConnected
    }

    private inner class NetworkStateReceiver : BroadcastReceiver() {

        override fun onReceive(context: Context, intent: Intent) {
            val isConnectedNow = isConnectedNow()

            for (listener in listeners) {
                listener.connectivityChanged(isConnectedNow)
            }
        }
    }

    interface OnConnectivityChangedListener {
        fun connectivityChanged(availableNow: Boolean)
    }
}